include/envoy/http/access_log.h

```diff
@@ -89,9 +91,10 @@ class RequestInfo {
   virtual std::chrono::milliseconds duration() const PURE;
 
   /**
-   * @return the failure reason for richer log experience.
+   * @return the failure reason for richer log experience, this can be represented as binary OR of
+   * FailureReason enum values.
    */
-  virtual FailureReason failureReason() const PURE;
+  virtual uint64_t failureReason() const PURE;
```

Comment:
Well, that's not too much work anyway, main concern around not to be backward compatible. This is def no problem for us, for others it's a simple config change as well, prob not too much to worry about.